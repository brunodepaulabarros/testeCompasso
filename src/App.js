import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import React from 'react';
import PokemonDetails from "./screens/PokemonDetails";
import PokemonList from "./screens/PokemonList";


const App = () => {
  const Stack = createStackNavigator();
  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="PokemonList"
        screenOptions={{
          headerShown: false,
        }}
      >
        <Stack.Screen
          name="PokemonList"
          component={PokemonList}
        />
        <Stack.Screen
          name="PokemonDetails"
          component={PokemonDetails}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
