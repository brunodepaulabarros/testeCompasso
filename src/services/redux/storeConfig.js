import { combineReducers, createStore } from 'redux';
import pokemonRequestReducer from "./reducers/pokemonRequestReducer";

const reducers = combineReducers({
    pokemonRequestReducer,
})

const storeConfig = () => {
    return createStore(reducers)
}

export default storeConfig;