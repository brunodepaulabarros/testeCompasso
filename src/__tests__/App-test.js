import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import App from '../App';
import storeConfig from '../services/redux/storeConfig';

const Store = storeConfig();
test('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={Store}>
      <App />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
