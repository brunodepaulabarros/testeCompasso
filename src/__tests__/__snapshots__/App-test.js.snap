// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#fff",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "flex-end",
                              "flexDirection": "row",
                              "height": 80,
                              "justifyContent": "space-between",
                              "marginBottom": 0,
                              "marginLeft": 15,
                              "marginRight": 0,
                              "marginTop": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 26,
                              "fontWeight": "700",
                            }
                          }
                        >
                          Pokedex
                        </Text>
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/pokeball.png",
                            }
                          }
                          style={
                            Object {
                              "height": 180,
                              "position": "absolute",
                              "right": -40,
                              "top": -40,
                              "width": 180,
                            }
                          }
                        />
                      </View>
                      <View>
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ddd",
                                "borderBottomWidth": 0,
                                "borderColor": "#ddd",
                                "borderRadius": 15,
                                "borderWidth": 2,
                                "height": 150,
                                "marginBottom": 15,
                                "marginLeft": 10,
                                "marginRight": 10,
                                "marginTop": 0,
                                "paddingBottom": 20,
                                "paddingLeft": 5,
                                "paddingRight": 5,
                                "paddingTop": 5,
                              },
                            ]
                          }
                        >
                          <View
                            onLayout={[Function]}
                          >
                            <View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "margin": 5,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "marginTop": 5,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 50,
                                          "height": 80,
                                          "width": 80,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ddd",
                                "borderBottomWidth": 0,
                                "borderColor": "#ddd",
                                "borderRadius": 15,
                                "borderWidth": 2,
                                "height": 150,
                                "marginBottom": 15,
                                "marginLeft": 10,
                                "marginRight": 10,
                                "marginTop": 0,
                                "paddingBottom": 20,
                                "paddingLeft": 5,
                                "paddingRight": 5,
                                "paddingTop": 5,
                              },
                            ]
                          }
                        >
                          <View
                            onLayout={[Function]}
                          >
                            <View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "margin": 5,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "marginTop": 5,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 50,
                                          "height": 80,
                                          "width": 80,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ddd",
                                "borderBottomWidth": 0,
                                "borderColor": "#ddd",
                                "borderRadius": 15,
                                "borderWidth": 2,
                                "height": 150,
                                "marginBottom": 15,
                                "marginLeft": 10,
                                "marginRight": 10,
                                "marginTop": 0,
                                "paddingBottom": 20,
                                "paddingLeft": 5,
                                "paddingRight": 5,
                                "paddingTop": 5,
                              },
                            ]
                          }
                        >
                          <View
                            onLayout={[Function]}
                          >
                            <View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "margin": 5,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "marginTop": 5,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 50,
                                          "height": 80,
                                          "width": 80,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ddd",
                                "borderBottomWidth": 0,
                                "borderColor": "#ddd",
                                "borderRadius": 15,
                                "borderWidth": 2,
                                "height": 150,
                                "marginBottom": 15,
                                "marginLeft": 10,
                                "marginRight": 10,
                                "marginTop": 0,
                                "paddingBottom": 20,
                                "paddingLeft": 5,
                                "paddingRight": 5,
                                "paddingTop": 5,
                              },
                            ]
                          }
                        >
                          <View
                            onLayout={[Function]}
                          >
                            <View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "margin": 5,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 4,
                                          "height": 20,
                                          "width": 120,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "position": "relative",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 4,
                                            "height": 20,
                                            "marginTop": 5,
                                            "width": 80,
                                          },
                                          Object {
                                            "backgroundColor": "#E1E9EE",
                                            "overflow": "hidden",
                                          },
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "position": "relative",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "borderRadius": 50,
                                          "height": 80,
                                          "width": 80,
                                        },
                                        Object {
                                          "backgroundColor": "#E1E9EE",
                                          "overflow": "hidden",
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "height": 64,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
